flask/application/routes.py

```diff
@@ -140,6 +140,7 @@ def index():
                         }
                         asyncio.run(get_hasura_connection_with_params(updateDataById, params))
 
+                        # nosemgrep:github.workflows.config.open-redirect
```

Comment:
@rashley-iqt 
this can be resolved with importing the following library:
```
from urllib.parse import urlparse
```

and replacing line 143 (nosemgrep comment) and 144 above with:
```
dash_app_url = request.url_root + current_app.config.get('DASH_URL_BASE') + view['id']
return redirect(urlparse(dash_app_url))
```

Reply:
that is a much more elegant solution.